src/service/device.js

```diff
@@ -162,48 +173,36 @@ const Device = GObject.registerClass({
 
     // FIXME: backend should do this stuff
     get encryption_info() {
-        let localCert = null;
-        let remoteCert = null;
-
         // Bluetooth connections have no certificate so we use the host address
         if (this.connection_type === 'bluetooth') {
             // TRANSLATORS: Bluetooth address for remote device
             return _('Bluetooth device at %s').format('???');
-
-        // If the device is connected use the certificate from the connection
-        } else if (this.connected) {
-            remoteCert = this.channel.peer_certificate;
-
-        // Otherwise pull it out of the settings
-        } else if (this.paired) {
-            remoteCert = Gio.TlsCertificate.new_from_pem(
-                this.settings.get_string('certificate-pem'),
-                -1
-            );
         }
 
         // FIXME: another ugly reach-around
-        let lanBackend;
-
-        if (this.service !== null)
-            lanBackend = this.service.manager.backends.get('lan');
-
-        if (lanBackend && lanBackend.certificate)
-            localCert = lanBackend.certificate;
-
-
-        let verificationKey = '';
-        if (localCert && remoteCert) {
-            let a = localCert.pubkey_der();
-            let b = remoteCert.pubkey_der();
-            if (a.compare(b) < 0)
-                [a, b] = [b, a]; // swap
-            const checksum = new GLib.Checksum(GLib.ChecksumType.SHA256);
-            checksum.update(a.toArray());
-            checksum.update(b.toArray());
-            verificationKey = checksum.get_string();
+        const localCert = this.service.manager.backends.get('lan')?.certificate;
+        const remoteCert = this.channel?.peer_certificate;
+        if (!localCert || !remoteCert)
+            return '';
+
+        const checksum = new GLib.Checksum(GLib.ChecksumType.SHA256);
+        let [a, b] = [localCert.pubkey_der(), remoteCert.pubkey_der()];
+        if (a.compare(b) < 0)
+            [a, b] = [b, a]; // swap
+        checksum.update(a.toArray());
+        checksum.update(b.toArray());
+
+        if (this.identity?.body.protocolVersion >= 8) {
+            const buffer = new ArrayBuffer(8);
+            const view = new DataView(buffer)
+                .setBigUint64(0, BigInt(this._pairingTimestamp));
+            checksum.update(new Uint8Array(buffer));
```

Comment:
I need to confirm this works correctly, since GSConnect tests against itself. Specifically, `setBigUint64()` and `BigInt()` affect the endianess of how the double is cast into the buffer.

There's a debug build for kdeconnect-android in the corresponding MR, I'm just having trouble getting it to install on my phone.